test/e2e/auth/service_accounts.go

```diff
@@ -418,6 +419,59 @@ var _ = SIGDescribe("ServiceAccounts", func() {
 		}
 	})
 
+	/*
+	  Release : v1.20
+	  Testname: TokenRequestProjection should mount a projected volume with token using TokenRequest API.
+	  Description: Ensure that projected service account token is mounted.
+	*/
+	ginkgo.It("should mount projected service account token when requested", func() {
+
+		var (
+			podName         = "test-pod-" + string(uuid.NewUUID())
+			volumeName      = "test-volume"
+			volumeMountPath = "/test-volume"
+			tokenVolumePath = "/test-volume/token"
+		)
+
+		volumes := []v1.Volume{
+			{
+				Name: volumeName,
+				VolumeSource: v1.VolumeSource{
+					Projected: &v1.ProjectedVolumeSource{
+						Sources: []v1.VolumeProjection{
+							{
+								ServiceAccountToken: &v1.ServiceAccountTokenProjection{
+									Path:              "token",
+									ExpirationSeconds: utilptr.Int64Ptr(60 * 60),
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+		volumeMounts := []v1.VolumeMount{
+			{
+				Name:      volumeName,
+				MountPath: volumeMountPath,
+				ReadOnly:  true,
+			},
+		}
+		mounttestArgs := []string{
+			"mounttest",
+			fmt.Sprintf("--file_content=%v", tokenVolumePath),
+		}
+
+		pod := e2epod.NewAgnhostPod(f.Namespace.Name, podName, volumes, volumeMounts, nil, mounttestArgs...)
+		pod.Spec.RestartPolicy = v1.RestartPolicyNever
+
+		output := []string{
+			fmt.Sprintf("content of file \"%v\": %s", tokenVolumePath, ".+"),
```

Comment:
`.+` isn't a very strong test... maybe something that shows it was a JWT?

Reply:
done